typebackend/typingviews.py

```diff
@@ -91,4 +129,24 @@ def get(self,request):
             para_yet_to_be_typed=Paragraph.objects.all()
         chosen_paragraph=choice(para_yet_to_be_typed)
         serializers=ParagraphSerializer(chosen_paragraph)
-        return Response(serializers.data)
\ No newline at end of file
+        return Response(serializers.data)
+
+class Dashboard(APIView):
+    permission_classes=[IsAuthenticated]
+
+    def get(self,request):
+        # { avg wpm, avg accuracy, user since, streak, Inactive days, longest streak, Total stresks, mode count }
+        user_id=request.user.id
+
+        dashboard_data={}
+        dashboard_data['user_since']=str(User.objects.get(id=user_id).date_joined)[0:10]
+        total_log=PractiseLog.objects.filter(user=user_id).count()
+        wpm_and_accuracy=PractiseLog.objects.filter(user_id=user_id).aggregate(wpm=Sum('wpm')/total_log,accuracy=Sum('accuracy')/total_log)
```

Comment:
Add this in Dashboard model too. These are persistent and expensive to calculate data.